File: Zoo_Java/src/Zoo/Plant.java
package Zoo;

import java.util.Random;

public abstract class Plant extends Organism {

    static public final int DefaultSpreadChance = 5;

    public Plant(World world) {
        super(world);
    }

    public Plant(World world, int x, int y) {
        super(world, x, y);
    }

    public Plant(World world, int strength, int initiative, int x, int y, int age) {
        super(world, strength, initiative, x, y, age);
    }

    @Override
    public abstract void takeAction();

    @Override
    public void collision(Organism attacker) {
        if (attacker.getStrength() > strength) {
            attacker.changePosition(x, y);
            world.logAttack(attacker, this);
            world.deleteOrganism(this);
        }
    }

    public void spread(Organism organism) {
        if (new Random().nextInt(DefaultSpreadChance) == 0) {
            int[] newPos = getRandomMovePosition(x, y);

            organism.setXAndY(newPos[0], newPos[1]);

            world.addOrganism(organism);
        }
    }
}

File: Zoo_Python/plant.py
from abc import abstractmethod
from random import random
from organism import Organism


class Plant(Organism):

    SPREAD_CHANCE = 0.2

    # def __init__(self, world):
    #     super().__init__(world)

    # def __init__(self, world, x, y):
    #     super().__init__(world, x, y)

    # def __init__(self, world, strength, initiative, x, y, age):
    #     super().__init__(world, strength, initiative, x, y, age)

    @abstractmethod
    def takeAction(self):
        pass

    def collision(self, attacker):
        if attacker.getStrength() > self._strength:
            attacker.changePosition( self.x, self.y )
            # log attack
            self._world.deleteOrganism(self)

    def spread( self, organism ):
        if random() <= self.SPREAD_CHANCE:
            newPos = self.getRandomMovePosition(self.x, self.y)
            organism.setXAndY(newPos[0], newPos[1])
            self._world.addOrganism(organism)
            
